TileLayer: default to strict-origin-when-cross-origin referrer-policy

IvanSanchez opened the pull request:
Sets the default value of the TileLayer referrerPolicy to `strict-origin-when-cross-origin` (which should already be the default anyway).

This is an alternative to https://github.com/Leaflet/Leaflet/pull/9883/

Fixes: #10156

IvanSanchez commented:
I feel a tiny bit slightly better setting the default to `strict-origin-when-cross-origin` for **all** tile domains instead of just for OSM. After all, as we discussed at #9883, this *should* be already the default.

cc @Firefishy @tomhughes @jonkoops @Falke-Design

jonkoops reviewed (commented):
Generally agree with just making this the default, just a couple of notes/questions in the implementation.

Falke-Design reviewed (dismissed):
I also prefer a array of url.
Also a warning about changing to https makes sense in my eyes

Falke-Design commented:
I prefer this over #9883

Falke-Design commented on src/layer/tile/TileLayer.js:
```diff
@@ -94,17 +94,15 @@ export class TileLayer extends GridLayer {
-
+			this._url = url.replace(/^http:\/\//, 'https://');
+    }
+    
```
```suggestion
		}
```

jonkoops commented on src/layer/tile/TileLayer.js:
```diff
@@ -94,14 +94,14 @@ export class TileLayer extends GridLayer {
+				if (options.attribution === null) {
+					options.attribution = '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors';
+				}
+				this._url = url.replace(/^http:\/\//, 'https://');
```
I can see if I can dedicate some time to this after the 2.x release. Right now I want to focus on getting a release candidate out of the window.

simon04 commented:
Hi @jonkoops, to me it seems that the discussion has converged towards using `new URL` (as performance is not an issue), and not using regex, right? So this PR could be finished rather easily...

jonkoops commented:
Yes, that would be my take.

Falke-Design committed:
TileLayer: default to strict-origin-when-cross-origin referrer-policy
TileLayer: automatically upgrade OSM tile URL templates to HTTPS
Using URL and array of hostnames

Falke-Design committed:
Add test

Falke-Design commented:
@IvanSanchez @simon04 @jonkoops please review again

mourner reviewed (approved):
This looks good to me, let's land this.
